ibm/resource_ibm_is_security_group_network_interface_attachment.go

```diff
@@ -183,17 +184,27 @@ func resourceIBMISSecurityGroupNetworkInterfaceAttachmentRead(d *schema.Resource
 	sgID := parts[0]
 	nicID := parts[1]
 
-	getSecurityGroupNetworkInterfaceOptions := &vpcv1.GetSecurityGroupNetworkInterfaceOptions{
+	getSecurityGroupNetworkInterfaceOptions := &vpcv1.GetSecurityGroupTargetOptions{
 		SecurityGroupID: &sgID,
 		ID:              &nicID,
 	}
-	instanceNic, response, err := sess.GetSecurityGroupNetworkInterface(getSecurityGroupNetworkInterfaceOptions)
+	secGroupTarget, response, err := sess.GetSecurityGroupTarget(getSecurityGroupNetworkInterfaceOptions)
 	if err != nil {
```

Comment:
add a condition if err != nil || secGroupTarget != nil  else if secGroupTarget is nil then will fail
instance_id := strings.Split(*secGroupTarget.(*vpcv1.SecurityGroupTargetReference).Href, "/")[5]